x-pack/platform/plugins/shared/streams_app/public/components/stream_list_view/documents_column.tsx

```diff
@@ -43,24 +43,28 @@ export function DocumentsColumn({
     },
   } = useKibana();
 
-  const { absoluteTimeRange } = useTimeFilter();
-  const minInterval = Math.floor((absoluteTimeRange.end - absoluteTimeRange.start) / numDataPoints);
+  const { timeState } = useTimefilter();
+
+  const minInterval = Math.floor((timeState.end - timeState.start) / numDataPoints);
 
   const histogramQueryFetch = useStreamsAppFetch(
-    async ({ signal }) => {
+    async ({ signal, timeState: { start, end } }) => {
       return streamsRepositoryClient.fetch('POST /internal/streams/esql', {
         params: {
           body: {
             operationName: 'get_doc_count_for_stream',
             query: `FROM ${indexPattern} | STATS doc_count = COUNT(*) BY @timestamp = BUCKET(@timestamp, ${minInterval} ms)`,
-            start: absoluteTimeRange.start,
-            end: absoluteTimeRange.end,
+            start,
+            end,
           },
         },
         signal,
       });
     },
-    [streamsRepositoryClient, indexPattern, absoluteTimeRange, minInterval]
+    [streamsRepositoryClient, indexPattern, , minInterval],
+    {
+      withTimeRange: true,
+    }
   );
```

Comment:
I'm not convinced this is simpler or clearer than what we had before here. 

Using a dependency array is a standard way of defining when a side effect should happen across React built in hooks and in the wider ecosystem. Having a mixed approach of implicit behaviour while still requiring the explicit dependency array feels messy and could cause confusion about when this side effect will get re-triggered, especially for anyone new to the codebase.

It also doesn't look like it reduces the amount of code we need to write or simplifies it in any way so I'm not so sure what the purpose of mixing the two concerns is. Data fetching and time filter state seem like two distinct concerns. Where do you see the overlap?

Reply:
The reason why I did this is because of a refresh of absolute time ranges. You cannot use epoch or iso timestamps then to invalidate dependencies. I do agree with mixing implicit vs explicit behaviour, but my guess is that as long as it's opt-in, and well-typed, it should be fine?

Reply:
if I select 2025-04-02T00:00:00.000Z until 2025-04-02T00:15:00.000Z, and press refresh, the data should still be refreshed

Reply:
but if you use those as dependencies for a hook, it won't be updated

Reply:
> if I select 2025-04-02T00:00:00.000Z until 2025-04-02T00:15:00.000Z, and press refresh, the data should still be refreshed

Should it? I think I'd want that combined into a single request

Reply:
Yes it should be refreshed, but maybe we are misunderstanding eachother. If the user has selected an absolute time range, and they click on refresh, what should happen to the data on the screen? Maybe another example: if they select "today", which rounds off the start and end times to last night and next night, what should happen if they click refresh?

Reply:
Oh, got you! You mean for an absolute time range that is not based off "now" at all. 
Good point that wouldn't change the calculated epoch currently. It should definitely still refresh in that case. 
I think I would still handle that using an explicit dependency list though. (e.g. `absoluteTimeRange` object would change and invalidate dependencies even if their `start` and `end` props remain the same)

Reply:
the risk there is IMHO that people will still use absoluteTimeRange.start and absoluteTimeRange.end as dependencies. I haven't really figured out how to express this well  in a way that it works intuitively w/ React hook dependencies. That is something I haven't solved, but I think at least for useStreamsAppFetch we can prevent it ourselves.